editor/components/NotesCard.tsx

```diff
@@ -0,0 +1,75 @@
+import Button from "react-bootstrap/Button";
+import { useState } from "react";
+import { FaCirclePlus } from "react-icons/fa6";
+import NotesModal from "./NotesModal";
+import RelatedRecordTable from "./RelatedRecordTable";
+import { ColDataCardDef } from "@/types/types";
+import AlignedLabel from "@/components/AlignedLabel";
+import DeleteNoteButton from "@/components/DeleteNoteButton";
+import PermissionsRequired from "@/components/PermissionsRequired";
+
+const allowedNoteRoles = ["vz-admin", "editor"];
+
+interface NotesCardProps<T extends Record<string, unknown>> {
+  notes: T[];
+  updateMutation: string;
+  insertMutation: string;
+  notesColumns: ColDataCardDef<T>[];
+  recordId: number | string;
+  onSaveCallback: () => Promise<void>;
+  refetch: () => Promise<void>;
+}
+
+const AddNoteButton = ({ onClick }: { onClick: () => void }) => {
+  return (
+    <PermissionsRequired allowedRoles={allowedNoteRoles}>
+      <Button size="sm" variant="primary" onClick={onClick}>
+        <AlignedLabel>
+          <FaCirclePlus className="me-2" />
+          Add note
+        </AlignedLabel>
+      </Button>
+    </PermissionsRequired>
+  );
+};
+
+/**
+ * UI component for adding a note to a crash
+ */
+export default function NotesCard<T extends Record<string, unknown>>({
+  notes,
+  notesColumns,
+  updateMutation,
+  insertMutation,
+  recordId,
+  onSaveCallback,
+}: NotesCardProps<T>) {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isValidating] = useState(false);
```

Comment:
might be missing something but this seems to be read-only. can `false` be passed directly to the prop below?

Reply:
good catch! i didnt write that code, but it seems u are right :thinking: